### PR TITLE
Update cross compile toolchains

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

`rbe_autoconfig` from bazel-toolchains has been deprecated since bazel 4 and is no longer supported. The canonical way is now to use the [rbe_configs_gen](https://github.com/bazelbuild/bazel-toolchains#rbe_configs_gen---cli-tool-to-generate-configs) CLI tool. This PR updates the cross compilation toolchains to use new configs generated by that CLI tool and updates the README instructions for maintaining these configs.

**Which issues(s) does this PR fix?**

This removes some annoying warnings that bazel was showing on every build.

**Other notes for review**

### CODE REVIEW

File: build/bazelrc/cross.bazelrc [hunk 1]
```diff
@@ -43,14 +43,10 @@ build:docker-sandbox --spawn_strategy=docker --strategy=Javac=docker --genrule_s
 build:docker-sandbox --define=EXECUTOR=remote
 build:docker-sandbox --experimental_docker_verbose
 build:docker-sandbox --experimental_enable_docker_sandbox
-build:docker-sandbox --crosstool_top=@rbe_ubuntu_clang//cc:toolchain 
-build:docker-sandbox --host_javabase=@rbe_ubuntu_clang//java:jdk 
-build:docker-sandbox --javabase=@rbe_ubuntu_clang//java:jdk 
-build:docker-sandbox --host_java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8 
-build:docker-sandbox --java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8
```
Comment (prestonvanloon) [inline build/bazelrc/cross.bazelrc:50]: These flags were deprecated in bazel 5.0.0. See https://github.com/bazelbuild/bazel/issues/7849

File: tools/cross-toolchain/README.md [hunk 2]
```diff
@@ -11,24 +11,63 @@ This package declares a c++ toolchain suite with cross compilers for targeting f
...
+|------------------|--------------------------------------------------------------------|--------------------------------------------|------------------------------------|------------------------------------|------------------------------------|
 | `//beacon-chain` | :heavy_check_mark:  docker-sandbox and RBE, supported locally only | :heavy_check_mark:  docker-sandbox and RBE | :heavy_check_mark:  docker-sandbox | :heavy_check_mark:  docker-sandbox | :heavy_check_mark:  docker-sandbox |
-| `//validator`    | :heavy_check_mark:  docker-sandbox and RBE | :heavy_check_mark: docker-sandbox and RBE | :heavy_check_mark:  docker-sandbox | :heavy_check_mark:  docker-sandbox | :heavy_check_mark:                 |
+| `//validator`    | :heavy_check_mark:  docker-sandbox and RBE                         | :heavy_check_mark: docker-sandbox and RBE  | :heavy_check_mark:  docker-sandbox | :heavy_check_mark:  docker-sandbox | :heavy_check_mark:                 |
```
Comment (prestonvanloon) [inline tools/cross-toolchain/README.md:59]: Formatting changes here, it looks good on widescreen, i promise!

File: tools/cross-toolchain/README.md [hunk 3]
```diff
@@ -11,24 +11,63 @@ This package declares a c++ toolchain suite with cross compilers for targeting f
...
-| Local run                     | `linux_amd64` | `linux_arm64` | `osx_amd64`               | `osx_arm64`               | `windows_amd64`              | 
-| Docker sandbox                | `linux_amd64_docker` | `linux_arm64_docker` | `osx_amd64_docker`        | `osx_arm64_docker`        | `windows_amd64_docker `      |
+| Config                        | linux_amd64                | linux_arm64                 | osx_amd64                 | osx_arm64                 | windows_amd64                 |
+|-------------------------------|----------------------------|-----------------------------|---------------------------|---------------------------|-------------------------------|
+| Local run                     | `linux_amd64`              | `linux_arm64`               | `osx_amd64`               | `osx_arm64`               | `windows_amd64`               |
+| Docker sandbox                | `linux_amd64_docker`       | `linux_arm64_docker`        | `osx_amd64_docker`        | `osx_arm64_docker`        | `windows_amd64_docker `       |
```
Comment (prestonvanloon) [inline tools/cross-toolchain/README.md:70]: Formatting changes here, it looks good on widescreen, i promise!

File: tools/cross-toolchain/configs/LICENSE [hunk 4]
```diff
@@ -0,0 +1,202 @@
+
+                                 Apache License
```
Comment (prestonvanloon) [inline tools/cross-toolchain/configs/LICENSE:2]: This Apache license was generated by the cli tool. I think it's OK to leave it.

**Comment by prestonvanloon** [comment]
Need to test cross compiled binaries work on the various platforms. Will test tomorrow.

**Review by rauljordan — APPROVED** [review]

**Pull request merged**